server/validate/email/validate.go

```diff
@@ -365,12 +271,12 @@ func (v *validator) Request(user t.Uid, email, lang, resp string, tmpToken []byt
 	base64.StdEncoding.Encode(token, tmpToken)
 
 	// Generate expected response as a random numeric string between 0 and 999999.
-	code, err := crand.Int(crand.Reader, big.NewInt(maxCodeValue))
+	code, err := crand.Int(crand.Reader, v.maxCodeValue)
 	if err != nil {
 		return false, err
 	}
 	resp = strconv.FormatInt(code.Int64(), 10)
-	resp = strings.Repeat("0", codeLength-len(resp)) + resp
+	resp = strings.Repeat("0", v.CodeLength-len(resp)) + resp
 
 	var template *textt.Template
 	if v.langMatcher != nil {
```

Comment:
Need fallback mechanism in case `lang` doesn't match any available templates.

Reply:
It's automatic. The matcher will use the first language from the list when all else fails.